bundles/org.eclipse.jface/src/org/eclipse/jface/preference/SpinnerFieldEditor.java

```diff
@@ -0,0 +1,370 @@
+/*******************************************************************************
+ * Copyright (c) 2003, 2015 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Wedia - Joel DRIGO (joel.drigo@wedia-group.com): Bug 470866
+ *     Sebastian Thomschke - Adapted code of ScaleFieldEditor for SpinnerFieldEditor
+ *******************************************************************************/
+package org.eclipse.jface.preference;
+
+import static org.eclipse.swt.events.SelectionListener.widgetSelectedAdapter;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Spinner;
+
+/**
+ * A field editor for an integer type preference. This class may be used as is,
+ * or subclassed as required.
+ *
+ * @since 3.35
+ */
+public class SpinnerFieldEditor extends FieldEditor {
+
+	/**
+	 * Value that will feed {@link Spinner#setDigits(int)}.
```

Comment:
These comments don't help me understand the code. 
```suggestion
	 * amount of digits to display (?) by default (?)
```